selectors/views/channel_sidebar.ts

```diff
@@ -1,60 +1,61 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+/* eslint-disable max-nested-callbacks */
+
 import {createSelector} from 'reselect';
 
-import {getUnreadChannelIds, getCurrentChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getAllChannels, getCurrentChannelId, getMyChannelMemberships, getUnreadChannelIds} from 'mattermost-redux/selectors/entities/channels';
 import {makeGetChannelsByCategory, makeGetCategoriesForTeam} from 'mattermost-redux/selectors/entities/channel_categories';
+import {getLastPostPerChannel} from 'mattermost-redux/selectors/entities/posts';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
 import {Channel} from 'mattermost-redux/types/channels';
 import {ChannelCategory} from 'mattermost-redux/types/channel_categories';
-import {RelationOneToOne} from 'mattermost-redux/types/utilities';
+import {isChannelMuted} from 'mattermost-redux/utils/channel_utils';
 
 import {getItemFromStorage} from 'selectors/storage';
 import {GlobalState} from 'types/store';
 import {StoragePrefixes} from 'utils/constants';
 import {getPrefix} from 'utils/storage_utils';
 
-export function isCategoryCollapsedFromStorage(prefix: string, storage: {[key: string]: any}, categoryId: string) {
+function isCategoryCollapsedFromStorage(prefix: string, storage: {[key: string]: any}, categoryId: string) {
     return getItemFromStorage(storage, prefix + StoragePrefixes.CHANNEL_CATEGORY_COLLAPSED + categoryId, false);
 }
 
 export function isCategoryCollapsed(state: GlobalState, categoryId: string) {
-    return isUnreadFilterEnabled(state) || isCategoryCollapsedFromStorage(getPrefix(state), state.storage.storage, categoryId);
+    return isCategoryCollapsedFromStorage(getPrefix(state), state.storage.storage, categoryId);
 }
 
 export function isUnreadFilterEnabled(state: GlobalState) {
     return state.views.channelSidebar.unreadFilterEnabled;
 }
 
-export function makeGetCollapsedStateForAllCategoriesByTeam(): (state: GlobalState, teamId: string) => Record<string, boolean> {
+// getChannelsInCategoryOrder returns an array of channels on the current team that are currently visible in the sidebar.
+// Channels are returned in the same order as in the sidebar.
+export const getChannelsInCategoryOrder = (() => {
```

Comment:
I realize that previously passing in the team ID as a parameter might have been good for letting us reuse selectors, but it also made them much more complicated since you had to worry about making a selector factory to avoid mucking up memoization.